sqlglot/dialects/databricks.py

```diff
@@ -51,6 +51,15 @@ class Generator(Spark.Generator):
             exp.ToChar: lambda self, e: self.function_fallback_sql(e),
         }
 
+        def columndef_sql(self, expression: exp.ColumnDef, sep: str = " ") -> str:
+            c = expression.find(exp.GeneratedAsIdentityColumnConstraint)
+            kind = self.sql(expression, "kind")
+            if c and isinstance(expression.this, exp.Identifier) and kind == "INT":
+                # only BIGINT generated identity constraints are supported
+                expression = expression.copy()
+                expression.set("kind", "BIGINT")
```

Comment:
I'd rename `c` to `constraint` or something. Also, what if it's a different kind of integer, besides `INT`?

Finally, a couple of general comments:
- Compare types using the `is_type` helper, e.g. `kind.is_type("int")`.
- Imitate the AST produced by `parse_one`, so `kind` here should be set to `exp.DataType.build("bigint")`.

Reply:
@GeorgeSittas 
If type is already BIGINT, nothing to do, if type is date or string or guid, then that's maybe way off semantically to coerce. If type is a SHORT or BYTE, which probably are probably not used. If INT64, INT128 those could be coerced to BIGINT... Thoughts?

Reply:
`kind` is a `str` and it does not have a method `.is_type()`
When searching, `this` and `expression` are used with `.is_type() helper`